src/mjolnir/valhalla_add_predicted_traffic.cc

```diff
@@ -276,164 +54,18 @@ int main(int argc, char** argv) {
-  std::unordered_map<GraphId, std::vector<std::string>> files_per_tile;
-  for (filesystem::recursive_directory_iterator i(traffic_tile_dir), end; i != end; ++i) {
-    if (i->is_regular_file()) {
-      // remove any extension
-      auto file_name = i->path().string();
-      auto pos = file_name.rfind(filesystem::path::preferred_separator);
-      file_name = file_name.substr(0, file_name.find('.', pos == std::string::npos ? 0 : pos));
-      try {
-        // parse it into a tile id and store the file path with it
-        auto id = GraphTile::GetTileId(file_name);
-        files_per_tile[id].push_back(i->path().string());
-      } catch (...) {}
-    }
-  }
-  std::vector<std::pair<GraphId, std::vector<std::string>>> traffic_tiles(files_per_tile.begin(),
-                                                                          files_per_tile.end());
-  std::random_device rd;
-  std::shuffle(traffic_tiles.begin(), traffic_tiles.end(), std::mt19937(rd()));
-
-  std::vector<std::shared_ptr<std::thread>> threads(config.get<uint32_t>("mjolnir.concurrency"));
-
-  LOG_INFO("Parsing speeds from " + std::to_string(traffic_tiles.size()) + " tiles.");
-  size_t floor = traffic_tiles.size() / threads.size();
-  size_t at_ceiling = traffic_tiles.size() - (threads.size() * floor);
-  decltype(traffic_tiles)::const_iterator tile_start, tile_end = traffic_tiles.begin();
+  // Prepare traffic tiles
+  auto traffic_tiles = vj::PrepareTrafficTiles(traffic_tile_dir);
+  // Get tile directory from config
   auto tile_dir = config.get<std::string>("mjolnir.tile_dir");
-  // A place to hold the results of those threads (exceptions, stats)
-  std::list<std::promise<stats>> results;
-  // Atomically pass around stats info
-  for (size_t i = 0; i < threads.size(); ++i) {
-    // Where the range begins
-    tile_start = tile_end;
-    // Where the range ends
-    tile_end += (i < at_ceiling ? floor + 1 : floor);
-    // Make the thread
-    results.emplace_back();
-    threads[i].reset(
-        new std::thread(update_tiles, tile_dir, tile_start, tile_end, std::ref(results.back())));
-  }
-
-  // wait for it to finish
-  for (auto& thread : threads)
-    thread->join();
-
-  // collect some stats
-  uint32_t constrained_count = 0, free_flow_count = 0, compressed_count = 0, updated_count = 0,
-           duplicate_count = 0;
-  for (auto& result : results) {
-    try {
-      auto thread_stats = result.get_future().get();
-      constrained_count += thread_stats.constrained_count;
-      free_flow_count += thread_stats.free_flow_count;
-      compressed_count += thread_stats.compressed_count;
-      updated_count += thread_stats.updated_count;
-      duplicate_count += thread_stats.dup_count;
-    } catch (std::exception& e) {
-      // TODO: throw further up the chain?
-    }
+  // Process traffic tiles
+  auto final_stats =
+      vj::ProcessTrafficTiles(tile_dir, traffic_tiles, config.get<uint32_t>("mjolnir.concurrency"));
```

Comment:
i wonder if this could be the only publicly exposed method and then inside of it has this little bit of logic we've left in this executable. this would allow us to make this method the only publicly visible one from the header file. then the rest of them can go straight into the traffic_updater.cc source file and the public api can only grow by one method instead of 7 or so. i know we didnt explain this in the beginning so no worries just clean up items to keep our lives easier in the future!